source/first-boot.rst

```diff
@@ -1,5 +1,46 @@
 First-Boot Customization
 ========================
 
-This is a placeholder for more documentation about first-boot
-customization.
+First-boot customization is performed using ``cloud-init``.
+``cloud-init`` runs during boot of all Ubuntu instances, discovers the
+environment in which it is running and configures the instance
+accordingly.  As well as configuring networking and user access, it can
+also perform customization based on instructions provided by the cloud
+vendor (vendor data) and by users (user data).  We are specifically
+interested in vendor-data.
+
+Vendor Data
+-----------
+
+Configuring Vendor Data
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Vendor data is presented to the instance as a plain text file.  How
+this file is configured depends on the implementation of your cloud,
+so you will need to determine how to configure it for your cloud.  For
```

Comment:
Wouldn't it be helpful for readers to know samples of this implementations?
If they are considering adding vendor data support, knowing how others have done it might be nice.
Not a blocker for this PR anyway.